File: csharp/perfect-numbers/PerfectNumbers.cs
using System;

public enum Classification
{
    Perfect,
    Abundant,
    Deficient
}

public static class PerfectNumbers
{
    public static Classification Classify(int number)
    {
        if (number <= 0)
        {
            throw new ArgumentOutOfRangeException("The number entered must be greater than 0.");
        }
        int aliquotSum = 0;
        for (int x = 1; x < number; x++)
        {
            if (number % x == 0)
            {
                aliquotSum += x;
            }
        }

        if (aliquotSum == number)
        {
            return Classification.Perfect;
        }
        else if (aliquotSum > number)
        {
            return Classification.Abundant;
        }
        else
        {
            return Classification.Deficient;
        }
        // throw new NotImplementedException();
    }
}

File: javascript/perfect-numbers/perfect-numbers.js
//
// This is only a SKELETON file for the 'Perfect Numbers' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const classify = (number) => {
  if (number <= 0){
    throw new Error('Classification is only possible for natural numbers.');
  }
  let sum = 0;
  for (let x = 1; x < number; x++){
      if (number % x == 0){
          sum += x;
      }
  }
  if (sum === number){return 'perfect';}
  else if (sum > number){return 'abundant';}
  else if (sum < number){return 'deficient';}
};
